packages/xo-web/src/common/stats.js

```diff
@@ -0,0 +1,75 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { forOwn } from 'lodash'
+
+import _ from './intl'
+import { fetchHostStats, fetchSrStats, fetchVmStats } from './xo'
+import { Select } from './form'
+
+const DEFAULT_GRANULARITY = 'seconds'
+
+const OPTIONS = [
+  {
+    label: _('statLastTenMinutes'),
+    value: 'seconds',
+  },
+  {
+    label: _('statLastTwoHours'),
+    value: 'minutes',
+  },
+  {
+    label: _('statLastDay'),
+    value: 'lastDay',
+  },
+  {
+    label: _('statLastWeek'),
+    value: 'hours',
+  },
+  {
+    label: _('statLastYear'),
+    value: 'days',
+  },
+]
+
+export const SelectGranularity = ({
+  onChange,
+  value = DEFAULT_GRANULARITY,
+  ...props
+}) => <Select {...props} onChange={onChange} options={OPTIONS} value={value} />
+
+SelectGranularity.propTypes = {
+  onChange: PropTypes.func.isRequired,
+  value: PropTypes.array.isRequired,
+}
+
+// ===================================================================
+
+const FETCH_FN_BY_TYPE = {
+  host: fetchHostStats,
+  sr: fetchSrStats,
+  vm: fetchVmStats,
+}
+
+const CUSTOMIZED_GRANULARITY = {
+  lastDay: {
+    granularity: 'hours',
+    nValues: 24,
+  },
```

Comment:
We can probably have all the info in one place:
```js
const OPTIONS = [
  {
    label: _('statLastDay'),
	value: 'lastDay',
    granularity: 'hours',
    keep: 24,
  },
  {
    label: _('statLastWeek'),
	value: 'lastWeek',
    granularity: 'hours',
  },
  ...
]
```
and splice the stats when the selected option has a `keep`.